pkg/kubectl/cmd/logs/logs.go

```diff
@@ -298,6 +302,47 @@ func (o LogsOptions) RunLogs() error {
 		return err
 	}
 
+	if o.Follow && len(requests) > 1 {
+		if len(requests) > o.MaxFollowConcurency {
+			return fmt.Errorf(
+				"you are attempting to follow %d log streams, but maximum allowed concurency is %d. Use --max-follow-concurency to increase the limit",
+				len(requests), o.MaxFollowConcurency,
+			)
+		}
+
+		return o.concurrentConsumeRequest(requests)
+	}
+
+	return o.sequentialConsumeRequest(requests)
+}
+
+func (o LogsOptions) concurrentConsumeRequest(requests []rest.ResponseWrapper) error {
+	piper, pipew := io.Pipe()
+	wg := &sync.WaitGroup{}
+	wg.Add(len(requests))
+	for _, request := range requests {
+		go func(request rest.ResponseWrapper) {
+			if err := o.ConsumeRequestFn(request, pipew); err != nil {
+				pipew.CloseWithError(err)
+
+				// It's important to return here to propagate the error via the pipe
+				return
+			}
+
+			wg.Done()
+		}(request)
+	}
+
+	go func() {
```

Comment:
I think you want to block the main flow until you hear back from the wait group, iow. this code should not be part of any goroutine.

Reply:
`io.Copy` below blocks the main flow. If we do not close pipe writer (`pipew` currently), it will be blocking the main flow forever (even if the server closed the connection). So we are waiting for all requests to finish in a separate goroutine and close the writer (as a result `io.Copy` stops blocking the main flow).

Or do I miss something?